### ISSUE TITLE
Authentication w/ Github API

### ISSUE THREAD

**Issue by malakada** [melissanoelle/hubflow#11]
Thanks @rebootd! <3

**Comment by rebootd** [comment]
yeah me... going to look at this.

**Comment by rebootd** [comment]
This is how we init and get github authorization:

``` javascript
OAuth.initialize('eVe8h6XL-bB0dAZ9eEZmHMbe-6Q')
OAuth.popup('github').done(function(result) {
    console.log(result)
    // do some stuff with result
});
```

**Comment by rebootd** [comment]
the result looks like this:

``` javascript
{ access_token: "<token>", 
token_type: "bearer", 
provider: "github"}
```

**Comment by rebootd** [comment]
This is how we need to call the issues api if we have an oauth'd user:
https://developer.github.com/guides/basics-of-authentication/#making-authenticated-requests

``` javascript
$.get("https://api.github.com/repos/" + this.state.repo + "/issues?state=all&access_token=<token>", function(result) {...
```

Or pass a header like this:

``` javascript
$.ajax({
      url: "https://api.github.com/repos/" + this.state.repo + "/issues?state=all",
      type: "GET",
      headers: headers,
      success: function(result) {
        if (this.isMounted()) {
          this.setState({
            issues: result,
          });
        }
      }.bind(this)
    });
```

**Comment by rebootd** [comment]
just added a branch that includes a request which optionally uses a github authToken if available.

**Comment by malakada** [comment]
Resolved in #15.
